independent-projects/qute/core/src/main/java/io/quarkus/qute/Location.java

```diff
@@ -0,0 +1,24 @@
+package io.quarkus.qute;
+
+import static java.lang.annotation.ElementType.FIELD;
+import static java.lang.annotation.ElementType.METHOD;
+import static java.lang.annotation.ElementType.PARAMETER;
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+
+/**
+ * Qualifies an injected template. The {@link #value()} is used to locate the template; it represents the path relative from
+ * the base path.
+ */
+@Retention(RUNTIME)
+@Target({ FIELD, PARAMETER, METHOD })
+public @interface Location {
```

Comment:
Are you sure about the name? Sounds a bit too generic, I can expect other libs will use the same name, no? What about `@TemplatePath`?

Reply:
Well, I'm not aware of any other `Location` annotation. The problem with `@TemplatePath` is that an injection point would look like `@TemplatePath("/foo/bar") Template bar`, i.e. too many templates. It also corresponds with the [TemplateLocator](https://github.com/quarkusio/quarkus/blob/master/independent-projects/qute/core/src/main/java/io/quarkus/qute/TemplateLocator.java) API.

Reply:
That said I'm not sure at all ;-)

Reply:
Me neither. Probably we should ask for feedback about this.

Reply:
`@TemplateFile`? :)

Reply:
`@Resource`?

Reply:
> `@Resource`?

`javax.annotation.Resource` and `jakarta.annotation.Resource`...

Reply:
:(

Reply:
`@TemplateLocation`?